modules/javafx.graphics/src/main/native-glass/mac/GlassMenu.m

```diff
@@ -262,6 +266,21 @@ - (void)_setPixels:(jobject)pixels
         (*env)->CallVoidMethod(env, pixels, jPixelsAttachData, ptr_to_jlong(&image));
         if (image != NULL)
         {
+            if ((jPixelsWidthField > 0)
+                && (jPixelsHeightField > 0)
+                && (jPixelsScaleXField > 0)
+                && (jPixelsScaleYField > 0)
+            ) {
+                jint width = (*env)->GetIntField(env, pixels, jPixelsWidthField);
+                jint height = (*env)->GetIntField(env, pixels, jPixelsHeightField);
+                jfloat scalex = (*env)->GetFloatField(env, pixels, jPixelsScaleXField);
+                jfloat scaley = (*env)->GetFloatField(env, pixels, jPixelsScaleYField);
+
+                if ((scalex > 1) && (scaley > 1) && (width > 1) && (height > 1)) {
```

Comment:
The check for `width` or `height` should be `> 0`, since a width or height of 1 is valid (perhaps not a likely value for an icon, but I see no reason to exclude it).

Also, in theory, `scalex` and `scaley` could be different such that one of them is `1` and the other is `> 1`, in which case you would still want to apply the scale. That might not be worth worrying about here (I doubt it can ever happen on macOS).